tensorflow/core/common_runtime/dml/dml_execution_context.h

```diff
@@ -125,78 +132,96 @@ class DmlExecutionContextImpl {
   void CloseCommandListAndExecute();
 };
 
-// A thread-safe wrapper over DmlExecutionContextImpl.
+// A thread-safe wrapper over DmlExecutionContextImpl. Calls to this class are
+// batched to minimize work within the lock, and the batched calls are
+// periodically flushed by a background thread (or by explicitly calling Flush).
 class DmlExecutionContext {
  public:
   DmlExecutionContext(ID3D12Device* d3d12_device, IDMLDevice* dml_device,
                       ID3D12CommandQueue* queue, DmlAllocator* allocator);
 
+  ~DmlExecutionContext();
+
   void Close() {
-    std::unique_lock<std::mutex> lock(mutex_);
-    impl_->Close();
+    std::unique_lock<std::mutex> lock(shared_state_->mutex);
+    shared_state_->impl->Close();
```

Comment:
It seems pretty hairy that this doesn't flush the pending batch(es). Is this ever called by anyone, given that our devices are now global singletons? If not, can we just delete all these Close() methods to avoid this hairiness?

Reply:
Already deleted all of these in a separate branch. It will come with some more refactoring.